ComponentKit/DataSources/CKCollectionViewDataSource.mm

```diff
@@ -73,6 +73,11 @@ - (instancetype)init
   CK_NOT_DESIGNATED_INITIALIZER();
 }
 
+- (void)dealloc
+{
+    _collectionView.dataSource = nil;
```

Comment:
I may be missing something, but `-[UICollectionView dataSource]` is defined as a weak pointer:

```objc
@property (nonatomic, weak, nullable) id <UICollectionViewDataSource> dataSource;
```

Which means that when this instance is deallocated, the reference to it will be nil-ed out. Isn't that right? Or am I misunderstanding this change? :question: